db.go

```
@@ -852,6 +852,39 @@ func (db *DB) CleanTombstones() (err error) {
 	return errors.Wrap(db.reload(), "reload blocks")
 }
 
+// LabelNames returns all the unique label names present in the DB in sorted order.
+func (db *DB) LabelNames() []string {
+	headIndexr, _ := db.head.Index()
+	labelNamesMap := make(map[string]struct{})
+
+	for _, name := range headIndexr.LabelNames() {
+		labelNamesMap[name] = struct{}{}
+	}
+
+	db.mtx.RLock()
+	for _, b := range db.blocks {
+		blockIndexr, err := b.Index()
+		if err != nil {
+			level.Error(db.logger).Log("msg", "block index reader fail", "err", err)
```

Specially here, we need to return this error rather than log it.